packages/terra-spacer/src/_spacerShorthandUtils.js

```diff
@@ -0,0 +1,99 @@
+import { SpacerSizes } from './_spacerConstants';
+
+const SHORTHAND_DELIMITER = ' ';
+const shLengthBoundaries = { min: 1, max: 4 };
+
+const conflictingPropNamesError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  conflictingPropName = '<SUPPLY CONFLICTING PROP NAME>',
+  componentName = 'Spacer',
+) => new Error(`Expected only ${propName} or ${conflictingPropName} to be supplied to ${componentName} but both were provided. Validation failed.`);
+
+const invalidShArgsLengthError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  valuesLength = '<SUPPLY RECEIVED ARGS LENGTH>',
+  minLength = shLengthBoundaries.min,
+  maxLength = shLengthBoundaries.max,
+  componentName = 'Spacer',
+) => new Error(`Invalid number of argument substrings provided to ${propName} in ${componentName}. Expected ${minLength}-${maxLength}. Received ${valuesLength}. Validation failed.`);
+
+const invalidShArgValuesError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  invalidShValues = [],
+  expectedValues = Object.values(SpacerSizes),
+  componentName = 'Spacer',
+) => new Error(`Expected ${propName} shorthand values to be one of ${expectedValues} but "${invalidShValues}" supplied to ${componentName}. Validation Failed.`);
+
+export const ShorthandPropErrors = {
```

Comment:
Where are these used at? I just see them used in the test code. If that's the case, I say remove this export, and do the error testing through the methods this file has.

Reply:
This grouping is just meant to serve as a binding for error generator functions. I use the encapsulated functions themselves within the [`shorthandValidator`](https://github.com/cerner/terra-core/blob/spacer-shorthand-enhancement/packages/terra-spacer/src/_spacerShorthandUtils.js#L57) to generate errors, and group them here into one export so that I can ensure equivalent error objects are generated during the appropriate tests. 

> ... remove this export, and do the error testing through the methods this file has.

can you elaborate?

The file itself only exports (and is only intended to export) `shorthandValidator` and `mapShorthandToObject`. 

In order to serve as a custom prop-types validator, `shorthandValidator` has to be defined as a single function, and rather than breaking each of the failing conditions checks/generators into their own top-level function, I'm taking advantage of closures to reduce the number of redundant calculations needed to test failing conditions, while encapsulating any required logic for each of the failing-condition checks into its own method